### ISSUE TITLE
403 ERROR: The Request could not be satisfied

### ISSUE THREAD

**Issue by Pratiksha1903** [cowinapi/developer.cowin#262]
I also used user agent but then only getting same error.
var headerObj={"User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64)"};
        
        	var apiResponse=https.get({
        		url:'https://cdn-api.co-vin.in/api/v2/appointment/sessions/public/calendarByDistrict?district_id=363&date=15-05-2021',
        		headers: headerObj
        	});

**Comment by saran-surya** [comment]
Hi @Pratiksha1903 , Is your server in INDIA ? or does the IP belong to INDIA ?
Was it in production or development ?

**Comment by heenahussain2** [comment]
I am having the same error 403 while trying to execute from the API documentation page and also while using the python requests library

**Comment by idevesh** [comment]
> I am having the same error 403 while trying to execute from the API documentation page and also while using the python requests library

@heenahusain96 It shows error while executing from there API documentation page. But it works fine on Application
What location your server is located?
Have you added Header in your requests?

**Comment by Pratiksha1903** [comment]
> Hi @Pratiksha1903 , Is your server in INDIA ? or does the IP belong to INDIA ?
> Was it in production or development ?

No...what was the steps to host the Indian server....now I am trying this code in netsuite.

**Comment by ShreePrince** [comment]
![image](https://user-images.githubusercontent.com/37611843/118623710-22d9eb00-b7e6-11eb-824b-306a27a912c9.png)

I'm getting the forbidden message when i execute the API's can anyone please help on this ??

**Comment by idevesh** [comment]
> ![image](https://user-images.githubusercontent.com/37611843/118623710-22d9eb00-b7e6-11eb-824b-306a27a912c9.png)
> 
> I'm getting the forbidden message when i execute the API's can anyone please help on this ??

Are you passing header value in Headers
If not pass it in the headers - 
User-Agent : Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36 Edg/90.0.818.51

**Comment by ShreePrince** [comment]
Yes i tried passing the User agent in the header, but no luck..

**Comment by saran-surya** [comment]
> > Hi @Pratiksha1903 , Is your server in INDIA ? or does the IP belong to INDIA ?
> > Was it in production or development ?
> 
> No...what was the steps to host the Indian server....now I am trying this code in netsuite.

The IP must be from INDIA to access the API's

**Comment by saran-surya** [comment]
> Yes i tried passing the User agent in the header, but no luck..

Did you try passing it with JSON.stringify()  or JSON.dumps in python ?

**Comment by ShreePrince** [comment]
> > Yes i tried passing the User agent in the header, but no luck..
> 
> Did you try passing it with JSON.stringify() or JSON.dumps in python ?


Actually im developing it in Javascript on ServiceNow platform, and it is the string which im passing.

**Comment by saran-surya** [comment]
> > > Yes i tried passing the User agent in the header, but no luck..
> > 
> > 
> > Did you try passing it with JSON.stringify() or JSON.dumps in python ?
> 
> Actually im developing it in Javascript on ServiceNow platform, and it is the string which im passing.

you should be passing a json, try using JSON.Stringify()

**Comment by prannoy-py** [comment]
Just pass these headers along with the params in your get request. Worked for me in python.

{
"Accept-Language": "en_US",
"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36 Edg/90.0.818.51"
}

**Comment by jaliph** [comment]
This API has to be tried from the INDIA location only. the Cloudfront gateway won't allow NON INDIA origin API calls

**Comment by PrashaanthT** [comment]
> I am having the same error 403 while trying to execute from the API documentation page and also while using the python requests library



> ![image](https://user-images.githubusercontent.com/37611843/118623710-22d9eb00-b7e6-11eb-824b-306a27a912c9.png)
> 
> I'm getting the forbidden message when i execute the API's can anyone please help on this ??

Were you able to resolve this?? I'm also facing the same in Snow

**Comment by PrashaanthT** [comment]
@ShreePrince @Pratiksha1903 were you guys able to resolve this? Im also facing the same issue in snow

**Comment by saran-surya** [comment]
Hey @PrashaanthT , if you are using a NODE JS server, you can try using the server proxy I created, I am currently using it in my Heroku app for the same purpose, maybe this can help,
[https://github.com/saran-surya/cassata](https://github.com/saran-surya/cassata)

Some people using it.
comment : [232](https://github.com/cowinapi/developer.cowin/issues/232#issuecomment-842854801)
